java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java

```diff
@@ -0,0 +1,59 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.scan;
+
+import java.util.ArrayList;
+import java.util.List;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * This base class represents the common set of optional arguments for the SCAN family of commands.
+ * Concrete implementations of this class are tied to specific SCAN commands (SCAN, HSCAN, SSCAN,
+ * and ZSCAN).
+ */
+@SuperBuilder
+public abstract class ScanOptions {
```

Comment:
I mentioned this in another PR too.  
Can we rename this BaseScanOptions please?  
Then I can use ScanOptions for the `scan` command.

Reply:
I did this rename in the HScan PR.

Reply:
I'll just do this work now in this PR and clean up the HScan PR after.